modules/formulize/class/durationElement.php

```diff
@@ -0,0 +1,271 @@
+				'max_minutes' => 0, // 0 means no maximum
+				'size' => 3 // width of each input box
+			);
+			$dataToSendToTemplate['ele_value'] = $ele_value;
+		}
+
+		return $dataToSendToTemplate;
+	}
+
+	// Save admin UI data
+	function adminSave($element, $ele_value)
+	{
+		if (is_object($element) && is_subclass_of($element, 'formulizeelement')) {
+			// Save which units to show
+			$ele_value['show_days'] = isset($_POST['show_days']) ? 1 : 0;
+			$ele_value['show_hours'] = isset($_POST['show_hours']) ? 1 : 0;
+			$ele_value['show_minutes'] = isset($_POST['show_minutes']) ? 1 : 0;
+
+			// Save min/max durations
+			$ele_value['min_minutes'] = intval($_POST['min_minutes']);
+			$ele_value['max_minutes'] = intval($_POST['max_minutes']);
+
+			// Save input box size
+			$ele_value['size'] = intval($_POST['size']);
+
+			$element->setVar('ele_value', $ele_value);
+
+			return true;
+		}
+		return false;
+	}
+
+	// Convert form input to minutes for storage
+	function prepareDataForSaving($value, $element, $entry_id = null)
+	{
+		if (!is_array($value)) {
+			return NULL;
+		}
+
+		$ele_value = $element->getVar('ele_value');
+
+		$totalMinutes = 0;
+
+		foreach ($this->timeUnits as $unit => $multiplier) {
+			if (isset($value[$unit]) && is_numeric($value[$unit])) {
+				$totalMinutes += $value[$unit] * $multiplier;
+			}
+		}
+
+		// Validate min/max duration
+		if ($ele_value['min_minutes'] > 0 && $totalMinutes < $ele_value['min_minutes']) {
+			return $ele_value['min_minutes'];
+		}
+		if ($ele_value['max_minutes'] > 0 && $totalMinutes > $ele_value['max_minutes']) {
+			return $ele_value['max_minutes'];
+		}
+
+		return $totalMinutes > 0 ? $totalMinutes : NULL;
+	}
+
+	function prepareDataForDataset($value, $handle, $entry_id) {
+		return $value;
+	}
+
+	// Convert stored minutes back to time units for display
+	function loadValue($value, $ele_value, $element)
+	{
+		if (!is_numeric($value)) {
+			return $ele_value;
+		}
+
+		$minutes = intval($value);
+		$breakdown = array();
+
+		// Convert minutes to time units
+		$remaining = $minutes;
+		foreach ($this->timeUnits as $unit => $multiplier) {
+			if ($ele_value['show_' . $unit]) {
+				$amount = floor($remaining / $multiplier);
+				$breakdown[$unit] = $amount;
+				$remaining -= ($amount * $multiplier);
+			}
+		}
+
+		$ele_value['values'] = $breakdown;
+		return $ele_value;
+	}
+
+	// Render the duration inputs
+	function render($ele_value, $caption, $markupName, $isDisabled, $element, $entry_id, $screen = false, $owner = null)
+	{
+		if ($isDisabled) {
+			// Render as text for disabled state
+			$output = "";
+			if (isset($ele_value['values'])) {
+				foreach ($ele_value['values'] as $unit => $amount) {
+					if ($amount > 0 && $ele_value['show_' . $unit]) {
+						$output .= $amount . " " . $unit . " ";
+					}
+				}
+			}
+			return new XoopsFormLabel($caption, trim($output), $markupName);
+		}
+
+		// Create container for inputs
+		$container = new XoopsFormElementTray($caption, ' ');
```

Comment:
Should the concatenator be a : instead of a blank space? Just a thought.

Reply:
Actually, I guess since the boxes are prefixed with the label of the unit, we don't want colons. Might be a nice option to add at some point, to use captions or show a X:X:X format. If you think that's easy enough / worthwhile enough, go for it now.

Reply:
If this is done, then it would affect the disabled display above too